language-plutus-core/src/Language/PlutusCore/Evaluation/MachineException.hs

```diff
@@ -31,22 +35,22 @@ data MachineException = MachineException
     , _machineExceptionCause :: Term TyName Name ()  -- ^ A 'Term' that caused the error.
     }
 
-instance PrettyCfg MachineError where
-    prettyCfg _   NonPrimitiveInstantiationMachineError =
+instance PrettyConstAppErrorBy config => PrettyBy config MachineError where
```

Comment:
Agreed. How about `ConstAppErrorPrettyBy`? It reads even weirder, but is not as misleading, I think. Or do you have a suggestion?

Reply:
What about `PrettyConstValueBy`? That's what it is, after all, it's only a coincidence that you sometimes use it for errors.

Reply:
I'd like to be explicit that it's needed for this specific type of errors. Otherwise we can just drop the constraint. Considering it has that weird name, I would be fine with dropping it.